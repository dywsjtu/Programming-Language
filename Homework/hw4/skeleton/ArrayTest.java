public class ArrayTest {
  /**
   *  Runs some tests on the Array class.
   */
  public static void main(String[] args) {
    // test 1D array
    Array<Integer> array1 = new Array<Integer>(10);
    System.out.println("rank: " + array1.rank());
    System.out.println("size: " + array1.size());
    System.out.println("shape: (" + array1.shape()[0] + ")");
    for (int i = 0; i < 10; i++) {
      array1.set(i, i);
    }
    for (int i = 0; i < 10; i++) {
      System.out.println("array1[" + i + "] = " + array1.get(i));
    }

    // test 2D array
    Array<Double> array2 = new Array<Double>(3, 4);
    System.out.println("rank: " + array2.rank());
    System.out.println("size: " + array2.size());
    System.out.println("shape: (" + array2.shape()[0] + "," +
                       array2.shape()[1] + ")");
    double counter2 = 0.1;
    for (int i = 0; i < 3; i++) {
      for (int j = 0; j < 4; j++) {
        array2.set(counter2++, i, j);
      }
    }
    for (int i = 0; i < 3; i++) {
      for (int j = 0; j < 4; j++) {
        System.out.println("array2[" + i + "," + j + "] = " +
                           array2.get(i, j));
      }
    }

    // test 3D array
    Array<String> array3 = new Array<String>(2, 3, 4);
    System.out.println("rank: " + array3.rank());
    System.out.println("size: " + array3.size());
    System.out.println("shape: (" + array3.shape()[0] + "," +
                       array3.shape()[1] + "," +
                       array3.shape()[2] + ")");
    int counter3 = 0;
    for (int i = 0; i < 2; i++) {
      for (int j = 0; j < 3; j++) {
        for (int k = 0; k < 4; k++) {
          array3.set("i" + counter3++, i, j, k);
        }
      }
    }
    for (int i = 0; i < 2; i++) {
      for (int j = 0; j < 3; j++) {
        for (int k = 0; k < 4; k++) {
          System.out.println("array3[" + i + "," + j + "," + k +
                             "] = " + array3.get(i, j, k));
        }
      }
    }
  }
}
