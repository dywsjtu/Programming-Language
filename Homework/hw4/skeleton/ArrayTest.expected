rank: 1
size: 10
shape: (10)
array1[0] = 0
array1[1] = 1
array1[2] = 2
array1[3] = 3
array1[4] = 4
array1[5] = 5
array1[6] = 6
array1[7] = 7
array1[8] = 8
array1[9] = 9
rank: 2
size: 12
shape: (3,4)
array2[0,0] = 0.1
array2[0,1] = 1.1
array2[0,2] = 2.1
array2[0,3] = 3.1
array2[1,0] = 4.1
array2[1,1] = 5.1
array2[1,2] = 6.1
array2[1,3] = 7.1
array2[2,0] = 8.1
array2[2,1] = 9.1
array2[2,2] = 10.1
array2[2,3] = 11.1
rank: 3
size: 24
shape: (2,3,4)
array3[0,0,0] = i0
array3[0,0,1] = i1
array3[0,0,2] = i2
array3[0,0,3] = i3
array3[0,1,0] = i4
array3[0,1,1] = i5
array3[0,1,2] = i6
array3[0,1,3] = i7
array3[0,2,0] = i8
array3[0,2,1] = i9
array3[0,2,2] = i10
array3[0,2,3] = i11
array3[1,0,0] = i12
array3[1,0,1] = i13
array3[1,0,2] = i14
array3[1,0,3] = i15
array3[1,1,0] = i16
array3[1,1,1] = i17
array3[1,1,2] = i18
array3[1,1,3] = i19
array3[1,2,0] = i20
array3[1,2,1] = i21
array3[1,2,2] = i22
array3[1,2,3] = i23
